File: src/Services/TalyerStudio.Customer/TalyerStudio.Customer.Application/Services/ServiceService.cs
using TalyerStudio.Customer.Application.Interfaces;
using TalyerStudio.Customer.Domain.Entities;
using TalyerStudio.Shared.Contracts.Services;

namespace TalyerStudio.Customer.Application.Services;

public class ServiceService : IServiceService
{
    private readonly IServiceRepository _repository;

    public ServiceService(IServiceRepository repository)
    {
        _repository = repository;
    }

    public async Task<ServiceDto?> GetByIdAsync(Guid id)
    {
        var service = await _repository.GetByIdAsync(id);
        return service == null ? null : MapToDto(service);
    }

    public async Task<List<ServiceDto>> GetAllAsync(Guid tenantId, Guid? categoryId = null, string? applicability = null, bool? isActive = null, int skip = 0, int take = 50)
    {
        var services = await _repository.GetAllAsync(tenantId, categoryId, applicability, isActive, skip, take);
        return services.Select(MapToDto).ToList();
    }

    public async Task<ServiceDto> CreateAsync(CreateServiceDto dto, Guid tenantId)
    {
        var service = await _repository.CreateAsync(dto, tenantId);
        return MapToDto(service);
    }

    public async Task<bool> UpdateAsync(Guid id, UpdateServiceDto dto)
    {
        var existingService = await _repository.GetByIdAsync(id);
        if (existingService == null)
        {
            throw new InvalidOperationException("Service not found");
        }

        return await _repository.UpdateAsync(id, dto);
    }

    public async Task<bool> DeleteAsync(Guid id)
    {
        return await _repository.DeleteAsync(id);
    }

    public async Task<List<ServiceDto>> GetByCategoryIdAsync(Guid categoryId)
    {
        var services = await _repository.GetByCategoryIdAsync(categoryId);
        return services.Select(MapToDto).ToList();
    }

    private ServiceDto MapToDto(Service service)
    {
        return new ServiceDto
        {
            Id = service.Id,
            TenantId = service.TenantId,
            Name = service.Name,
            Description = service.Description,
            CategoryId = service.CategoryId,
            CategoryName = service.Category?.Name,
            BasePrice = service.BasePrice,
            Currency = service.Currency,
            Applicability = service.Applicability.ToString(),
            EstimatedDurationMinutes = service.EstimatedDurationMinutes,
            IsActive = service.IsActive,
            DisplayOrder = service.DisplayOrder,
            Icon = service.Icon,
            CreatedAt = service.CreatedAt
        };
    }
}

File: src/Clients/talyerstudio-dashboard/src/services/serviceService.ts
import api, { API_URLS } from './api';

export interface Service {
  id: string;
  tenantId: string;
  name: string;
  description?: string;
  categoryId: string;
  categoryName?: string;
  basePrice: number;
  currency: string;
  applicability: string;
  estimatedDurationMinutes?: number;
  isActive: boolean;
  displayOrder?: number;
  createdAt: string;
  updatedAt?: string;
}

export interface ServiceCategory {
  id: string;
  tenantId: string;
  name: string;
  description?: string;
  icon?: string;
  displayOrder?: number;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateServiceRequest {
  tenantId: string;
  name: string;
  description?: string;
  categoryId: string;
  basePrice: number;
  currency: string;
  applicability: string;
  estimatedDurationMinutes?: number;
  isActive: boolean;
  displayOrder?: number;
}

export interface UpdateServiceRequest extends CreateServiceRequest {
  id: string;
}

export interface CreateServiceCategoryRequest {
  tenantId: string;
  name: string;
  description?: string;
  icon?: string;
  displayOrder?: number;
  isActive: boolean;
}

export interface UpdateServiceCategoryRequest extends CreateServiceCategoryRequest {
  id: string;
}

const serviceService = {
  // Services
  async getAll(): Promise<Service[]> {
    const response = await api.get(`${API_URLS.CUSTOMER}/services`);
    return response.data;
  },

  async getById(id: string): Promise<Service> {
    const response = await api.get(`${API_URLS.CUSTOMER}/services/${id}`);
    return response.data;
  },

  async create(data: CreateServiceRequest): Promise<Service> {
    const response = await api.post(`${API_URLS.CUSTOMER}/services`, data);
    return response.data;
  },

  async update(id: string, data: UpdateServiceRequest): Promise<Service> {
    const response = await api.put(`${API_URLS.CUSTOMER}/services/${id}`, data);
    return response.data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`${API_URLS.CUSTOMER}/services/${id}`);
  },

  // Service Categories
  async getCategories(): Promise<ServiceCategory[]> {
    const response = await api.get(`${API_URLS.CUSTOMER}/servicecategories`);
    return response.data;
  },

  async getCategoryById(id: string): Promise<ServiceCategory> {
    const response = await api.get(`${API_URLS.CUSTOMER}/servicecategories/${id}`);
    return response.data;
  },

  async createCategory(data: CreateServiceCategoryRequest): Promise<ServiceCategory> {
    const response = await api.post(`${API_URLS.CUSTOMER}/servicecategories`, data);
    return response.data;
  },

  async updateCategory(id: string, data: UpdateServiceCategoryRequest): Promise<ServiceCategory> {
    const response = await api.put(`${API_URLS.CUSTOMER}/servicecategories/${id}`, data);
    return response.data;
  },

  async deleteCategory(id: string): Promise<void> {
    await api.delete(`${API_URLS.CUSTOMER}/servicecategories/${id}`);
  }
};

export default serviceService;